server/tabular.js

```diff
@@ -112,16 +113,32 @@ Meteor.publish("tabular_getInfo", function(tableName, selector, sort, skip, limi
 
   var filteredCursor = table.collection.find(selector, findOptions);
 
-  var filteredRecordIds = filteredCursor.map(function (doc) {
-    return doc._id;
-  });
+  // var filteredRecordIds = filteredCursor.map(function (doc) {
+  //   return doc._id;
+  // });
 
   var countCursor = table.collection.find(selector, {fields: {_id: 1}});
 
   var recordReady = false;
+  var lastUpdatedRecords = 0;
   function updateRecords() {
+    var currTime = new Date().getTime();
+    if (throttleRefresh && currTime - lastUpdatedRecords < throttleRefresh) {
+      return;
+    }
+    else {
+      lastUpdatedRecords = currTime;
+    }
+
     var currentCount = countCursor.count();
 
+    // recalculate filtered records from scratch
+    // because we may have pushed/removed beyond the limit
+    // TODO: compare the length to the limit?
```

Comment:
Since `filteredCursor` already had the `limit` option passed, I don't think comparing would be necessary.